### ISSUE TITLE
Fold output when running under travis

### ISSUE THREAD

**Issue by thedrow** [eisensheng/pytest-catchlog#5]
Some times a failing test has too many logs and we'd like to see the failure and that's it.
Folding output can be done by printing fold markers e.g.:

```
travis_fold:start:<section>.<test name>.logs
INFO: bla bla
travis_fold:end:<section>.<test name>.logs
```

**Comment by eisensheng** [comment]
I thought about this feature for a while and I must say that it sounds useful. I'm also using travis-ci so I can see where this comes in handy but I must say that I don't see pytest-catchlog as the proper location for this functionality. 

It would also be nice to fold other sections like these for stderr and stdout for example. This would go beyond the responsibility of this plugin though. Thus IMHO this functionality would deserve its own plugin. Please tell me how you think about this. Thanks!

**Comment by The-Compiler** [comment]
I agree with @eisensheng here, this would best be in a separate plugin. I don't know if it's possible to do that with the hooks exposed by pytest though.

**Comment by abusalimov** [comment]
I believe this should be implemented as a separate plugin that would fold stdout/stderr as well. I actually even started to code that some time ago, but didn't try to run it yet, i.e. that's only intended to be a proof-of-concepts, just my thoughts on the issue. Please check out the [gist](https://gist.github.com/abusalimov/fc114197c40485aff660).

**Comment by thedrow** [comment]
I have no idea how to implement this as a separate plugin. Is there a hook that this plugin provides that I can use?

**Comment by abusalimov** [comment]
Meet [pytest-travis-fold](https://github.com/abusalimov/pytest-travis-fold)!

Add `pip install pytest-travis-fold` step to `.travis.yml` and it will fold captured stdout/stderr for failing tests. Folding also captured logs will be available once #7 is merged and a new version of pytest-catchlog is uploaded to PyPI.

**Comment by The-Compiler** [comment]
Nice, thanks for that @abusalimov! I added it to my build, and will see how it looks whenever a test fails next time :wink:

**Comment by abusalimov** [comment]
@The-Compiler You're very welcome!

**Comment by thedrow** [comment]
@abusalimov Thanks! will it fold logs from catchlog?

**Comment by The-Compiler** [comment]
@thedrow 

>  Folding also captured logs will be available once #7 is merged and a new version of pytest-catchlog is uploaded to PyPI.

**Comment by eisensheng** [comment]
Considering pytest-travis-fold and the fact that #7 finally was merged, I'll close this issue here since it appears to be resolved from my point of view.
